Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
+name = "scoped-mutex"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+critical-section = { version = "1.1", optional = true }
+
+[features]
+default = [
+    "impl-critical-section",
+]
+impl-critical-section = ["dep:critical-section"]
```

Comment:
why not just make this an optional dependency, and use `#[cfg(feature = "critical-section")]` in code?

Reply:
i've been avoiding dep names, but I don't have a good reason.